vnext/ReactUWP/Modules/LinkingManagerModule.cpp

```diff
@@ -53,10 +53,45 @@ canOpenURLAsync(winrt::Windows::Foundation::Uri uri, Callback success, Callback
 // LinkingManagerModule
 //
 const char *LinkingManagerModule::name = "LinkingManager";
+/*static*/ std::mutex LinkingManagerModule::s_mutex;
+/*static*/ Uri LinkingManagerModule::s_initialUri{nullptr};
+/*static*/ std::vector<LinkingManagerModule *> LinkingManagerModule::s_linkingModules;
+
+LinkingManagerModule::LinkingManagerModule() noexcept {
+  std::scoped_lock lock{s_mutex};
+  s_linkingModules.push_back(this);
+}
+
+LinkingManagerModule::~LinkingManagerModule() noexcept {
+  std::scoped_lock lock{s_mutex};
+  auto it = std::find(s_linkingModules.begin(), s_linkingModules.end(), this);
+  if (it != s_linkingModules.end()) {
+    s_linkingModules.erase(it);
+  }
+}
+
+/*static*/ void LinkingManagerModule::OpenUri(winrt::Windows::Foundation::Uri const &uri) noexcept {
+  if (!s_initialUri) {
+    s_initialUri = uri;
+  }
 
-LinkingManagerModule::LinkingManagerModule() {}
+  std::vector<LinkingManagerModule *> modules;
+  {
+    std::scoped_lock lock{s_mutex};
+    modules = s_linkingModules;
+  }
 
-LinkingManagerModule::~LinkingManagerModule() = default;
+  for (auto &module : modules) {
```

Comment:
I think there's technically a race condition here. I.e. if a module is destroyed directly after we copy the global list of modules, we will have a pointer to a now invalid module. 

Maybe you could hold the lock longer or reference count the module? Neither of those sound great either though I guess.

Reply:
You are right, let me address it in another PR. To avoid the race we must guarantee that the destructor is called in the main thread as this method does.

---
In reply to: [396197186](https://github.com/microsoft/react-native-windows/pull/4392#discussion_r396197186) [](ancestors = 396197186)